File: app/src/main/java/com/angogasapps/myfamily/ui/screens/registeractivity/EnterCodeFragment.java
package com.angogasapps.myfamily.ui.screens.registeractivity;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.angogasapps.myfamily.databinding.FragmentEnterCodeBinding;
import com.google.firebase.auth.PhoneAuthCredential;
import com.google.firebase.auth.PhoneAuthProvider;

import static com.angogasapps.myfamily.firebase.AuthFunctions.trySignInWithCredential;


public class EnterCodeFragment extends Fragment {
    private FragmentEnterCodeBinding binding;

    private String mPhoneNumber;
    private String id;

    public EnterCodeFragment(String mPhoneNumber, String id){
        this.mPhoneNumber = mPhoneNumber;
        this.id = id;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentEnterCodeBinding.inflate(getLayoutInflater(), container, false);

        return binding.getRoot();
    }

    @Override
    public void onStart() {
        super.onStart();
        //Прослушиваем набор текста пользователем
        binding.codeEditText.addTextChangedListener(new TextWatcher() {
            @Override
            // после набота текста
            public void afterTextChanged(Editable s) {
                String code = binding.codeEditText.getText().toString();
                if (code.length() == 6){
                    enterCode();
                }
            }
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {}
            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {}
        });
    }
    // функция для отправки и проверки введённого пользователем кода
    private void enterCode(){
        String code = binding.codeEditText.getText().toString();
        PhoneAuthCredential credential = PhoneAuthProvider.getCredential(id, code);
        trySignInWithCredential(getActivity(), credential);
    }
}

File: app/src/main/java/com/angogasapps/myfamily/ui/screens/registeractivity/EnterCodeFragment.kt
package com.angogasapps.myfamily.ui.screens.registeractivity

import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.angogasapps.myfamily.databinding.FragmentEnterCodeBinding
import com.angogasapps.myfamily.firebase.AuthFunctions
import com.google.firebase.auth.PhoneAuthProvider

class EnterCodeFragment(private val mPhoneNumber: String, private val id: String,
                        val onNewUserSignIn: () -> Unit, val onOldUserSignIn: () -> Unit) : Fragment() {

    private lateinit var binding: FragmentEnterCodeBinding
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View {
        binding = FragmentEnterCodeBinding.inflate(layoutInflater, container, false)
        return binding.root
    }

    override fun onStart() {
        super.onStart()
        binding.codeEditText.addTextChangedListener(object : TextWatcher {
            override fun afterTextChanged(s: Editable) {
                val code = binding.codeEditText.text.toString()
                if (code.length == 6) {
                    enterCode()
                }
            }

            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}
        })
    }

    private fun enterCode() {
        val code = binding.codeEditText.text.toString()
        val credential = PhoneAuthProvider.getCredential(id, code)
        AuthFunctions.trySignInWithCredential(activity, credential, onNewUserSignIn = onNewUserSignIn, onOldUserSignIn = onOldUserSignIn)
    }
}